plugin/trino-opensearch/src/test/java/io/trino/plugin/opensearch/BaseOpenSearchConnectorTest.java

```diff
@@ -0,0 +1,1954 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+public abstract class BaseOpenSearchConnectorTest
+        extends BaseConnectorTest
+{
+    private final String image;
+    private final String catalogName;
+    private OpenSearchServer opensearch;
+    protected RestHighLevelClient client;
+
+    BaseOpenSearchConnectorTest(String image, String catalogName)
+    {
+        this.image = image;
+        this.catalogName = catalogName;
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        opensearch = new OpenSearchServer(image, false, ImmutableMap.of());
+        HostAndPort address = opensearch.getAddress();
+        client = new RestHighLevelClient(RestClient.builder(new HttpHost(address.getHost(), address.getPort())));
+
+        return createOpenSearchQueryRunner(
+                opensearch.getAddress(),
+                TpchTable.getTables(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                3,
+                catalogName);
+    }
+
+    @AfterAll
+    public final void destroy()
+            throws IOException
+    {
+        opensearch.stop();
+        opensearch = null;
+        client.close();
+        client = null;
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        return switch (connectorBehavior) {
+            case SUPPORTS_ADD_COLUMN,
+                    SUPPORTS_COMMENT_ON_COLUMN,
+                    SUPPORTS_COMMENT_ON_TABLE,
+                    SUPPORTS_CREATE_MATERIALIZED_VIEW,
+                    SUPPORTS_CREATE_SCHEMA,
+                    SUPPORTS_CREATE_TABLE,
+                    SUPPORTS_CREATE_VIEW,
+                    SUPPORTS_DELETE,
+                    SUPPORTS_INSERT,
+                    SUPPORTS_LIMIT_PUSHDOWN,
+                    SUPPORTS_MERGE,
+                    SUPPORTS_RENAME_COLUMN,
+                    SUPPORTS_RENAME_TABLE,
+                    SUPPORTS_ROW_TYPE,
+                    SUPPORTS_SET_COLUMN_TYPE,
+                    SUPPORTS_TOPN_PUSHDOWN,
+                    SUPPORTS_UPDATE -> false;
+            default -> super.hasBehavior(connectorBehavior);
+        };
+    }
+
+    /**
+     * This method overrides the default values used for the data provider
+     * of the test {@link AbstractTestQueries#testLargeIn()} by taking
+     * into account that by default OpenSearch 2.x supports only up to `1024`
+     * clauses in query.
+     * <p>
+     * Consult `index.query.bool.max_clause_count` opensearch.yml setting
+     * for more details.
+     *
+     * @return the amount of clauses to be used in large queries
+     */
+    @Override
+    protected List<Integer> largeInValuesCountData()
+    {
+        return ImmutableList.of(200, 500, 1000);
+    }
```

Comment:
This is a connector limitation and should not be solved at the test level.
That was exactly the point of `testLargeIn` -- to ensure that the connector does not push predicates the remote system cannot process and that even more challenging queries do succeed. (even more important with DF).

This needs to be fixed.